scipy/signal/_short_time_fft.py

```diff
@@ -0,0 +1,1659 @@
+    ...         xlim=(t_lo, t_hi))
+    ...
+    >>> im1 = ax1.imshow(abs(Sx), origin='lower', aspect='auto',
+    ...                  extent=SFT.extent(N), cmap='viridis')
+    >>> ax1.plot(t_x, f_i, 'r--', alpha=.5, label='$f_i(t)$')
+    >>> fig1.colorbar(im1, label="Magnitude $|S_x(t, f)|$")
+    ...
+    >>> # Shade areas where window slices stick out to the side:
+    >>> for t0_, t1_ in [(t_lo, SFT.lower_border_end[0] * SFT.T),
+    ...                  (SFT.upper_border_begin(N)[0] * SFT.T, t_hi)]:
+    ...     ax1.axvspan(t0_, t1_, color='w', linewidth=0, alpha=.2)
+    >>> for t_ in [0, N * SFT.T]:  # mark signal borders with vertical line:
+    ...     ax1.axvline(t_, color='y', linestyle='--', alpha=0.5)
+    >>> ax1.legend()
+    >>> fig1.tight_layout()
+    >>> plt.show()
+
+    Reconstructing the signal with the `istft` is straightforward, but note
+    that the length of `x1` should be specified, since the SFT length
+    increases in `hop` steps:
+
+    >>> SFT.invertible  # check if invertible
+    True
+    >>> x1 = SFT.istft(Sx, k1=N)
+    >>> np.allclose(x, x1)
+    True
+
+    It is possible to calcluate the SFT of signal parts:
+
+    >>> p_q = SFT.nearest_k_p(N // 2)
+    >>> Sx0 = SFT.stft(x[:p_q])
+    >>> Sx1 = SFT.stft(x[p_q:])
+
+    When assembling sequential STFT parts together, the overlap needs to be
+    considered:
+
+    >>> p0_ub = SFT.upper_border_begin(p_q)[1] - SFT.p_min
+    >>> p1_le = SFT.lower_border_end[1] - SFT.p_min
+    >>> Sx01 = np.hstack((Sx0[:, :p0_ub],
+    ...                   Sx0[:, p0_ub:] + Sx1[:, :p1_le],
+    ...                   Sx1[:, p1_le:]))
+    >>> np.allclose(Sx01, Sx)  # Compare with SFT of complete signal
+    True
+
+    It is also possible to calculate the `itsft` for signal parts:
+
+    >>> y_p = SFT.istft(Sx, N//3, N//2)
+    >>> np.allclose(y_p, x[N//3:N//2])
+    True
+
+    """
+    # immutable attributes (only have getters but no setters):
+    _win: NDArray  # window
+    _dual_win: Optional[NDArray] = None  # canonical dual window
+    _hop: int  # Step of STFT in number of samples
+
+    # mutable attributes:
+    _fs: float  # sampling frequency of input signal and window
+    _fft_typ: FFT_TYP_TYPE = 'onesided'  # Type of FFT to use
+    _mfft: int  # length of FFT used - defaults to len(win)
+    _scaling: Optional[Literal['magnitude', 'psd']] = None  # Scaling of _win
+    _phase_shift: Optional[int]  # amount to shift phase of FFT in samples
+
+    # attributes for caching calculated values:
+    _fac_mag: Optional[float] = None
+    _fac_psd: Optional[float] = None
+    _lower_border_end: Optional[tuple[int, int]] = None
+
+    def __init__(self, win: NDArray, hop: int, fs: float,
+                 fft_typ: FFT_TYP_TYPE = 'onesided',
+                 mfft: Optional[int] = None,
+                 dual_win: Optional[NDArray] = None,
+                 scale_to: Optional[Literal['magnitude', 'psd']] = None,
+                 phase_shift: Optional[int] = 0):
```

Comment:
Good Point. Renamed `fft_type` to `fft_mode` (left the `fft_` part in, since `mode` is too generic and already used in the legacy implementation). In commit 126e8309a7f40ede82d5ea208f7dac178e659e51, I added the `*` parameter to `__init__`, `from_dual`, `from_window`, `stft`, `stft_detrend`, `spectrogram`, and `istft`.